databases/week1/mySQL-exercises.sql

```diff
@@ -0,0 +1,61 @@
+-- Design queries that retrieve the following data sets.
+
+-- Find out how many tasks are in the task table
+
+SELECT COUNT(*) FROM task;
+
+-- Find out how many tasks in the task table do not have a valid due date
+
+SELECT COUNT(*) FROM task WHERE due_date IS NULL;
+
+-- Find all the tasks that are marked as done
+
+SELECT * FROM task
+JOIN status ON task.status_id = status.id
+WHERE status.name = 'Done';
+
+-- Find all the tasks that are not marked as done
+
+SELECT * FROM task
+JOIN status ON task.status_id = status.id
+WHERE status.name <> 'Done';
+
+-- Get all the tasks, sorted with the most recently created first
+
+SELECT * FROM task 
+ORDER BY created DESC;
+
+-- Get the single most recently created task
+
+SELECT * FROM task
+ORDER BY created DESC
+LIMIT 1;
+
+-- Get the title and due date of all tasks
+ -- where the title or description contains database
+
+SELECT title, due_date FROM task
+WHERE title LIKE '%database%' || description LIKE '%database%';
+
+
+-- Get the title and status (as text) of all tasks
+
+SELECT title, status.name FROM task
+JOIN status ON task.status_id = status.id;
+
+-- Get the name of each status, along with a count of how many tasks have that status
+
+SELECT count(*), name
+FROM status
+JOIN task ON status.id = task.status_id
```

Comment:
Even though this query works for the set of data we have, what would happen if we add an extra status named `Postponed`? Also, how could we solve the problem?

Reply:
Hello Dan, I think I am having hard time imagining the case of extra status. Why do you think it would change how query works? Thank you :)

Reply:
Hey @bengu-thon-mai-mochi
The reason it will not work properly with this query in the case of an extra status is because this type of join (inner join) will only select the rows where the join condition is met. So because there is no task that has the extra status, the condition won't be met for that status.

It can even happen when we have no tasks with status the `Done` for example. Instead of showing
`Done = 0` we skip it completely.

I will explain it to you in class with some examples if it doesn't make sense. 🙌